labgrid/driver/shelldriver.py

```diff
@@ -60,7 +60,7 @@ def on_activate(self):
             self._await_login()
             self._inject_run()
         if self.keyfile:
-            self._put_ssh_key(self.keyfile)
+            self._put_ssh_key(self.target.env.config.resolve_path(self.keyfile))
```

Comment:
You need to handle the case of not having an environment. Grep for 'target.env' to see how this is done in other drivers.

Reply:
Fixed.